packages/core/src/lib/utils.mts

```diff
@@ -16,3 +16,18 @@ export const breakLines = (content: string, width: number): string =>
         .map((str) => str.trimEnd()),
     )
     .join('\n');
+
+/**
+ * Creates a 0-based index out of an integer, wrapping around if necessary.
+ */
+export const index = (max: number) => (value: number) => ((value % max) + max) % max;
+
+/**
+ * Rotates an array of items by an integer number of positions.
+ */
+export const rotate =
```

Comment:
Let's simplify those 2 functions by removing the currying. Let's make them a function with multiple arguments.